undertow/src/main/java/org/wildfly/extension/undertow/ActiveRequestTrackerRemove.java

```diff
@@ -0,0 +1,20 @@
+package org.wildfly.extension.undertow;
+
+import org.jboss.as.controller.AbstractRemoveStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceName;
+
+public class ActiveRequestTrackerRemove extends AbstractRemoveStepHandler {
```

Comment:
This operation handler is missing rollback behavior.

Reply:
I'm not sure what you're asking for here. Point to an example?

Reply:
If this operation is part of a batch, say operation 2 of 5, and a failure occurs in operation 3, 4 or 5, then the AbstractAddStepHandler.rollbackRuntime(...) method will be invoked for the purpose of undoing the work that was done in performRuntime(...).  Similarly, the AbstractRemoveStepHandler invokes its recoverServices(...) method for the purpose of undoing the work that was done in its performRemove(...) method.

Check out the generic AddOperationHandler and RemoveOperationHandler classes used by the clustering subsystems for an example:
https://github.com/wildfly/wildfly/blob/main/clustering/common/src/main/java/org/jboss/as/clustering/controller/AddStepHandler.java
https://github.com/wildfly/wildfly/blob/main/clustering/common/src/main/java/org/jboss/as/clustering/controller/RemoveStepHandler.java

N.B. The clustering subsystems encapsulate resource runtime behavior within a separate ResourceServiceHandler abstraction, since, generally speaking, the rollback logic of the add operation is the same as the runtime logic of the remove operation, and visa versa.